### PR TITLE
MM-29608 - Fix for RHS List across teams

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- This changes how the redux works -- it's now a map to a map: `{teamId: {channelId: Incident}}`. True, we could keep it all in a `{channelId: Incident}` flatter map, but I think that makes unnecessary waste in the selectors. The downside is the complicated reducer state spreading to avoid mutating the store. Would appreciate a double check if you're able.
- I fixed how we were using reselectors--it's making more sense to me now.
- Oh! I'm a huge fan of TDD with cypress--it helped catch a huge bug in an earlier version of this (I was sending a message to clear the store when we changed channels, but that solution is racy -- this current implementation isn't).

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-29608

### CODE REVIEW

**Review by agarciamontoro — APPROVED** [review]
Great! I left a couple of nits, but nothing blocking :tada:

Reducers look good, by the way, I like the map to map approach, and I don't find the state spreading that confusing. Also, `inIncident` is much more clear now, great addition!

**Review by lieut-data — APPROVED** [review]
Nice and crisp!

File: webapp/src/selectors.ts [hunk 1]
```diff
@@ -19,17 +21,27 @@ export const isIncidentRHSOpen = (state: GlobalState): boolean => pluginState(st
...
+        if (!incidentMapByTeam[teamId]) {
+            return [];
+        }
+
         const incidents = [] as Incident[];
-        for (const incident of Object.values(channelIdToIncidents)) {
+        for (const incident of Object.values(incidentMapByTeam[teamId])) {
             if (incident.is_active) {
                 incidents.push(incident);
             }
```
[the remark this answers is not in the document]
Comment (cpoile) [reply webapp/src/selectors.ts:47]: @agarciamontoro I'm not sure there is a `.values()' function available on a plain object -- can you double check?

File: tests-e2e/cypress/integration/frontstage/incident_rhs_list_spec.js [hunk 2]
```diff
@@ -320,6 +338,99 @@ describe('rhs incident list', () => {
...
+            // # start first incident
+            const now = Date.now();
+            const incidentName1 = 'Private ' + now;
+            cy.apiStartIncident({
+                teamId,
+                playbookId,
+                incidentName: incidentName1,
+                commanderUserId: userId
+            });
+            cy.verifyIncidentActive(teamId, incidentName1);
```
[the remark this answers is not in the document]
Comment (cpoile) [reply tests-e2e/cypress/integration/frontstage/incident_rhs_list_spec.js:366]: We do it after each time we call `apiStartIncident` -- Jesse started doing it, I'm following that pattern. I'm assuming it's a double check, which doesn't hurt. :)

File: webapp/src/selectors.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (agarciamontoro) [reply webapp/src/selectors.ts:47]: Oh shoot, I assumed it was a `Map`. We still can do something similar:
```js
return Object.values(incidentMapByTeam[teamId]).
    filter(i => i.is_active).
    sort((a, b) => b.create_at - a.create_at);
```
What do you think? I feel it's a bit more readable, but feel free to leave it as it is!

File: tests-e2e/cypress/integration/frontstage/incident_rhs_list_spec.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (agarciamontoro) [reply tests-e2e/cypress/integration/frontstage/incident_rhs_list_spec.js:367]: Sounds good!

File: webapp/src/selectors.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (cpoile) [reply webapp/src/selectors.ts]: Sure, that looks clean, made the change. Thanks!

[the remark this answers is not in the document]
Comment (agarciamontoro) [reply webapp/src/selectors.ts]: Thank you! :)

**Review by agarciamontoro — APPROVED** [review]
:rocket:

**Pull request merged**